### ISSUE TITLE
Whitespace in project directory breaks `devbox shellenv ...`

### ISSUE THREAD

**Issue by andrew-monroe** [jetify-com/devbox#2374]
### What happened?

# TLDR
[This line of code](https://github.com/jetify-com/devbox/blob/e8763dd86c11fabafceac47c16731fe0b3292d21/internal/devbox/devbox.go#L376C1-L377C1) is not doing proper whitespace escaping on the value of `d.projectDir()`. The issue may also be fixable by updating `shellgen.ScriptPath()`, but you all will know better than I.

# Description
`devbox shellenv --init-hook --install --no-refresh-alias` breaks when the project directory contains whitespace.

I specifically encounter this when I am running `devbox generate direnv --print-envrc`, which appears to be calling out to the above command on my system.

The reason this _ultimately_ fails is because this script produces this line following all of the `exports`:
```sh
. /Users/andrewmonroe/Documents/10-19 Workspace/10 App/canarytech_dev/.devbox/gen/scripts/.hooks.sh;
```

Which naturally leads to this error when the output code is executed:
```sh
/bin/bash:195: /Users/andrewmonroe/Documents/10-19: No such file or directory
```

### Steps to reproduce

1. Make a project within a directory whose path includes whitespace.
2. Run `devbox shellenv --init-hook --install --no-refresh-alias`.
3. See that the attempt to invoke the `.hooks.sh` file will not include whitespace-escaping.

### Command

shellenv

### Devbox version

0.13.5

### What system does this bug occur on?

macOS (Apple Silicon)

**Comment by andrew-monroe** [comment]
Hey @savil, just checking in! Any chance you can say when you'll be able to pick this up? Hoping it should turn out to be as easy as a one-line fix.

If needed I am also happy to attempt to put up a PR here if that would speed things up. I haven't written much go code in the past, but this seems pretty straightforward if you're open to outside contributions.

**Comment by savil** [comment]
@andrew-monroe thanks for the ping. This slipped past my radar. Sending a quick fix.

**Comment by andrew-monroe** [comment]
Ah, thanks so much @savil !! Much appreciated :)
